diff --git a/node_modules/@maniac-tech/react-native-expo-read-sms/android/build.gradle b/node_modules/@maniac-tech/react-native-expo-read-sms/android/build.gradle
index 6cef1b4..2827093 100644
--- a/node_modules/@maniac-tech/react-native-expo-read-sms/android/build.gradle
+++ b/node_modules/@maniac-tech/react-native-expo-read-sms/android/build.gradle
@@ -17,7 +17,7 @@ android {
     namespace "com.reactlibrary"
     defaultConfig {
         compileSdk 34
-        minSdkVersion 34
+        minSdkVersion 24
         targetSdkVersion 34
         versionCode 1
         versionName "1.0"
diff --git a/node_modules/@maniac-tech/react-native-expo-read-sms/android/src/main/java/com/reactlibrary/RNExpoReadSmsModule.java b/node_modules/@maniac-tech/react-native-expo-read-sms/android/src/main/java/com/reactlibrary/RNExpoReadSmsModule.java
index f78f1fe..97c29db 100644
--- a/node_modules/@maniac-tech/react-native-expo-read-sms/android/src/main/java/com/reactlibrary/RNExpoReadSmsModule.java
+++ b/node_modules/@maniac-tech/react-native-expo-read-sms/android/src/main/java/com/reactlibrary/RNExpoReadSmsModule.java
@@ -7,6 +7,7 @@ import android.content.Context;
 import android.content.Intent;
 import android.content.IntentFilter;
 import android.content.pm.PackageManager;
+import android.os.Build;
 import android.os.Bundle;
 import android.telephony.SmsMessage;
 import android.util.Log;
@@ -50,7 +51,7 @@ public class RNExpoReadSmsModule extends ReactContextBaseJavaModule {
           }
         };
         String SMS_RECEIVED_ACTION = "android.provider.Telephony.SMS_RECEIVED";
-        if(Build.VERSION.SDK_INT >= 34 && getApplicationInfo().targetSdkVersion >= 34) {
+        if(Build.VERSION.SDK_INT >= 34 && reactContext.getApplicationInfo().targetSdkVersion >= 34) {
           reactContext.registerReceiver(msgReceiver, new IntentFilter(SMS_RECEIVED_ACTION), Context.RECEIVER_EXPORTED);
         } else {
           reactContext.registerReceiver(msgReceiver, new IntentFilter(SMS_RECEIVED_ACTION));
